script.js

```diff
@@ -1,29 +1,64 @@
 function setup() {
   const allEpisodes = getAllEpisodes();
   makePageForEpisodes(allEpisodes);
+  searchEpisodeCards();
 }
 
 function makePageForEpisodes(allEpisodes) {
   const rootElem = document.getElementById("root");
-  const episodeList =  allEpisodes.map(createEpisodeCard)
-  rootElem.append(...episodeList)
+  const episodeList = allEpisodes.map(createEpisodeCard);
+  rootElem.append(...episodeList);
 }
 
-function formatSeason(season){
-  return season  < 10 ? `S0${season}` : `S${season}`
+function formatSeason(season) {
+  return season < 10 ? `S0${season}` : `S${season}`;
 }
 
-function formatEpisode(episode){
-  return episode  < 10 ? `E0${episode}` : `E${episode}`
+function formatEpisode(episode) {
+  return episode < 10 ? `E0${episode}` : `E${episode}`;
 }
 
-function createEpisodeCard(episode){
-  const episodeCard = document.querySelector('#episode-card-template').content.cloneNode(true)
-  episodeCard.querySelector('h1').textContent = episode.name
-  episodeCard.querySelector('[data-season-episode]').textContent = `- ${formatSeason(episode.season)}${formatEpisode(episode.number)}`
-  episodeCard.querySelector('img').setAttribute('src', episode.image.medium)
-  episodeCard.querySelector('p').textContent = episode.summary.replace(/<p>|<\/p>/g, '')
-  return episodeCard  
+function createEpisodeCard(episode) {
+  const episodeCard = document
+    .querySelector("#episode-card-template")
+    .content.cloneNode(true);
+  episodeCard.querySelector("h1").textContent = episode.name;
+  episodeCard.querySelector(
+    "[data-season-episode]"
+  ).textContent = `- ${formatSeason(episode.season)}${formatEpisode(
+    episode.number
+  )}`;
+  episodeCard.querySelector("img").setAttribute("src", episode.image.medium);
+  episodeCard.querySelector("p").textContent = episode.summary.replace(
+    /<p>|<\/p>/g,
+    ""
+  );
+  return episodeCard;
 }
 
+function searchEpisodeCards(allEpisodes) {
+  const input = document.getElementById("search-input");
+  const cards = document.querySelectorAll(".episode-card");
+  const displayNumber = document.getElementById("display-number");
+
+  input.addEventListener("input", (e) => {
+    const searchText = e.target.value.toLowerCase();
+    let countShownEpisodes = 0;
+
+    cards.forEach((card) => {
+      const cardText = card.textContent;
```

Comment:
What do you think might happen if the `toLowerCase()` method is not applied to the `cardText` variable, unlike the `searchText` variable?

Reply:
It is a mistake there should not be  ```javascript toLowerCase() ``` at all.

Reply:
OK if yo you are not using the ```toLowerCase()``` method, how are you handling the case-insensitive search? for example if you search using the term ```coming```, it should display the the first episode even though the match is ```Coming``` with capital C?

Reply:
I just thought to left it to be case sensitive but your example make sense so I agree that changing it and add  ```toLowerCase() ```method .